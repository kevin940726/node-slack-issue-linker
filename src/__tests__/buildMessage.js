const mockedLoadEnv = jest.fn(() => ({
  HOST: 'github',
  OWNER: 'owner',
  REPO: 'repo',
}));
jest.mock('../loadEnv', () => mockedLoadEnv);

const buildMessage = require('../buildMessage');

it('should give default values', () => {
  const matches = [{ issue: 12 }];

  expect(buildMessage(matches)).toMatchSnapshot();
});

it('should take match values', () => {
  const matches = [{ issue: 12, owner: 'kevin940726', repo: 'node-slack-issue-linker' }];

  expect(buildMessage(matches)).toMatchSnapshot();
});

it('should handle multiple matches', () => {
  const matches = [
    { issue: 12 },
    { issue: 3, repo: 'node-slack-issue-linker' },
    { issue: 6, owner: 'kevin940726', repo: 'node-slack-issue-linker' },
  ];

  expect(buildMessage(matches)).toMatchSnapshot();
});

it('should handle set HOST value', () => {
  mockedLoadEnv.mockImplementationOnce(() => ({
    HOST: 'https://gitlab.com',
  }));
  jest.resetModules();
  const mockedBuildMessage = require('../buildMessage');

  const matches = [{ issue: 12, owner: 'kevin940726', repo: 'node-slack-issue-linker' }];

  expect(mockedBuildMessage(matches)).toMatchSnapshot();
});
