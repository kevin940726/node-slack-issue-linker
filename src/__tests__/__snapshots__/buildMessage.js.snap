// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should give default values 1`] = `"<https://github.com/owner/repo/issues/12|repo#12> :point_right: _https://github.com/owner/repo/issues/12_"`;

exports[`should handle multiple matches 1`] = `
"<https://github.com/owner/repo/issues/12|repo#12> :point_right: _https://github.com/owner/repo/issues/12_
<https://github.com/owner/node-slack-issue-linker/issues/3|node-slack-issue-linker#3> :point_right: _https://github.com/owner/node-slack-issue-linker/issues/3_
<https://github.com/kevin940726/node-slack-issue-linker/issues/6|kevin940726/node-slack-issue-linker#6> :point_right: _https://github.com/kevin940726/node-slack-issue-linker/issues/6_"
`;

exports[`should handle set HOST value 1`] = `"<https://gitlab.com/kevin940726/node-slack-issue-linker/issues/12|kevin940726/node-slack-issue-linker#12> :point_right: _https://gitlab.com/kevin940726/node-slack-issue-linker/issues/12_"`;

exports[`should take match values 1`] = `"<https://github.com/kevin940726/node-slack-issue-linker/issues/12|kevin940726/node-slack-issue-linker#12> :point_right: _https://github.com/kevin940726/node-slack-issue-linker/issues/12_"`;
